elyra/validation/validate.py

```diff
@@ -0,0 +1,424 @@
+                    response.add_message(severity='1',
+                                         message_type="invalidNodeProperty",
+                                         message="Node field is incorrect type",
+                                         data={"nodeID": node['id'],
+                                               "propertyName": field_list})
+
+            if 'label' not in node_data['ui_data']:
+                response.add_message(severity='1',
+                                     message_type="invalidNodeProperty",
+                                     message="Node is missing field",
+                                     data={"nodeID": node['id'],
+                                           "propertyName": 'label'})
+            elif not isinstance(node_data['ui_data']['label'], str):
+                response.add_message(severity='1',
+                                     message_type="invalidNodeProperty",
+                                     message="Node field is incorrect type",
+                                     data={"nodeID": node['id'],
+                                           "propertyName": 'label'})
+
+            # Validate actual node property values
+            self.validate_filepath(node_id=node['id'], root_dir=root_dir, property_name='filename',
+                                   filename=node_data['filename'], response=response)
+
+            # If the running locally, we can skip the resource and image name checks
+            if pipeline_json['pipelines'][0]['app_data']['runtime'] != 'local':
+                self.validate_container_image_name(node, response=response)
+                self.validate_resource_value(node, resource_name='cpu', response=response)
+                self.validate_resource_value(node, resource_name='gpu', response=response)
+                self.validate_resource_value(node, resource_name='memory', response=response)
+            for dependency in node_data['dependencies']:
+                self.validate_filepath(node_id=node['id'], root_dir=root_dir, property_name='dependencies',
+                                       filename=dependency, response=response)
+            for env_var in node_data['env_vars']:
+                self.validate_environmental_variables(node_id=node['id'], env_var=env_var, response=response)
+            if node_data['filename'] != node_data['ui_data']['label']:
+                self.validate_ui_data_label(node_id=node['id'], label_name=node_data['ui_data']['label'],
+                                            response=response)
+
+    def validate_container_image_name(self, node, response):
+        """
+        Validates the image name exists and is proper in syntax
+        :param node: the node definition to be validated
+        :param response: ValidationResponse containing the issue list to be updated
+        """
+        image_name = node['app_data'].get('runtime_image', '')
+        if not image_name:
```

Comment:
Does this need to tolerate 'local' where images are not required?

Reply:
Never mind - I see this (and the resource validators) are conditional and only called when not 'local'.  Sorry I didn't notice that prevously.